docs/concepts/security.md

```diff
@@ -0,0 +1,48 @@
+# Security overview
+
+By default all Resoto components communicate with each other using TLS (https). The trust between components is established using a Pre-Shared-Key (PSK) that is used to derive a key which in turn is used to sign JWT tokens. `resotocore` is running a Public Key Infrastructure (PKI) including a Certificate Authority (CA). Upon start all components will request a certificate from the CA.
+
+## Pre-Shared-Key (PSK)
+Every component takes a `--psk` flag (which can alternatively be supplied using the environment variables `RESOTOCORE_PSK`, `RESOTOWORKER_PSK`, `RESOTOMETRICS_PSK`, `RESOTOSHELL_PSK`). The value of the flag is a pre-shared key. A common passphrase that all components know about. From this passphrase [a 256 bit key is derived](https://github.com/someengineering/resoto/blob/main/resotolib/resotolib/jwt.py) using a random salt. This key is used to sign JWT tokens.
+
+### JSON Web Tokens (JWT)
+If `--psk` was specified ever request that a component makes to `resotocore` must provide a valid `Authentication` header with a JWT signed using the PSK. This is true for encrypted https as well as unencrypted http requests. Meaning even if TLS is turned off (`--no-tls`) but a PSK was specified, the request will still be authenticated (but not encrypted!).
+
+## Certificate Authority (CA)
+
+### Establishing trust
+`resotocore` has two API endpoints, `/ca/cert` and `/ca/sign`. The former serves the core's public CA root certificate. Upon startup all other components will request this root certificate from the core without validating the https certificate. However, the core encodes the certificate's SHA256 fingerprint into a JWT which is signed with the previously mentioned PSK. When the component downloads the thus far untrusted root certificate it compares its fingerprint with the one the core has encoded in the JWT. If the fingerprints match the CA's root certificate is stored as a valid root certificate and the trust is established.
```

Comment:
I think we should mention the different basic setup modes:
- ca is created by resotocore, all components create their own certificate (default)
- certificate is provided to all components (manual setup)
- combination of the 2